### PR TITLE
Add get pending invitation API and OrganizationEvent type structure

### PR DESCRIPTION
**Pull request by kucuny**
I added team pending invitation API and OrganizationEvent type structures.

Team pending invitation API : https://developer.github.com/v3/orgs/teams/#list-pending-team-invitations

Organization Event Type : https://developer.github.com/v3/activity/events/types/#organizationevent

### CODE REVIEW

File: github/event_types.go [hunk 1]
```diff
@@ -276,6 +276,27 @@ type MilestoneEvent struct {
 	Org     *Organization `json:"organization,omitempty"`
 }
 
+// OrganizationEvent is Triggered when a user is added, removed, or invited to an Organization.
```
Comment (shurcooL) [inline github/event_types.go]: Nitpick, "Triggered" and "Organization" should be lower case.

File: github/event_types.go [hunk 2]
```diff
@@ -276,6 +276,27 @@ type MilestoneEvent struct {
...
+// Events of this type are not visible in timelines. These events are only used to trigger organization hooks.
+// Webhook event name is "organization"
```
Comment (shurcooL) [inline github/event_types.go]: Nitpick, missing period at end of sentence. Here, and in many of the places below.

File: github/orgs_teams.go [hunk 3]
```diff
@@ -377,3 +393,28 @@ func (s *OrganizationsService) RemoveTeamMembership(team int, user string) (*Res
...
+func (s *OrganizationsService) ListPendingTeamInvitations(team int, opt *ListOptions) ([]*Invitation, *Response, error) {
+	u := fmt.Sprintf("teams/%v/invitations", team)
+	u, err := addOptions(u, opt)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req.Header.Set("Accept", mediaTypeOrgMembershipPreview)
```
Comment (shurcooL) [inline github/orgs_teams.go]: Please add a `// TODO: remove custom Accept header when this API fully launches.` comment here, just like all other instances.

File: github/orgs_teams.go [hunk 4]
```diff
@@ -37,10 +40,23 @@ type Team struct {
...
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+}
+
 func (t Team) String() string {
 	return Stringify(t)
 }
 
+func (i Invitation) String() string {
+	return Stringify(i)
+}
```
Comment (shurcooL) [inline github/orgs_teams.go]: I think it'll be more readable and consistent to keep the type and its `String` method together. For example, see [here](https://github.com/google/go-github/blob/b272bb85ab53b610b5660988998feb0e0294bd2b/github/admin.go#L17-L60).

File: github/orgs_teams.go [hunk 5]
```diff
@@ -377,3 +393,28 @@ func (s *OrganizationsService) RemoveTeamMembership(team int, user string) (*Res
...
 }
+
+// ListPendingTeamInvitations get pending invitaion list in team.
+// Warning: The API may change without advance notice during the preview period.
+// Preview features are not supported for production use.
+//
+// GitHub API docs: https://developer.github.com/v3/orgs/teams/#list-pending-team-invitations
+func (s *OrganizationsService) ListPendingTeamInvitations(team int, opt *ListOptions) ([]*Invitation, *Response, error) {
+	u := fmt.Sprintf("teams/%v/invitations", team)
+	u, err := addOptions(u, opt)
```
Comment (shurcooL) [inline github/orgs_teams.go]: You're forgetting an error check here, after `addOptions`.

```Go
u, err := addOptions(u, opt)
if err != nil {
	return nil, nil, err
}
```

**Comment by kucuny** [comment]
@shurcooL Thanks for review my codes!

File: github/orgs_teams.go (hunk 5 above)
Comment (kucuny) [inline github/orgs_teams.go]: I added here:
https://github.com/google/go-github/pull/504/commits/102d95064c15a2cb23c68d1e7f72f1152c206f3f#diff-41b996921c268a5244f4ae6747c6b76dR405

File: github/orgs_teams.go [hunk 6]
```diff
@@ -41,6 +44,19 @@ func (t Team) String() string {
 	return Stringify(t)
 }
 
+// Invitation represents a team member's inviation status
+type Invitation struct {
+	ID        *int       `json:"id,omitempty"`
+	Login     *string    `json:"login,omitempty"`
+	Email     *string    `json:"email,omitempty"`
+	Role      *string    `json:"role,omitempty"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
```
Comment (shurcooL) [inline github/orgs_teams.go]: What's the source for the `created_at` field? I don't see it in the example at https://developer.github.com/v3/activity/events/types/#organizationevent.

Comment (kucuny) [inline github/orgs_teams.go]: @shurcooL 
FYI https://developer.github.com/changes/2016-12-20-organization-team-api-update/

You can see the github announced api change log.
So, maybe github will update api doc. :)

Comment (shurcooL) [inline github/orgs_teams.go]: Perfect, thanks for answering my question!

File: github/orgs_teams_test.go [hunk 7]
```diff
@@ -499,3 +499,25 @@ func TestOrganizationsService_ListUserTeams(t *testing.T) {
...
 	}
 }
+
+func TestOrganizationsService_ListPendingTeamInvitations(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/teams/1/invitations", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testFormValues(t, r, values{"page": "1"})
```
Comment (gmlewis) [inline github/orgs_teams_test.go]: Please add a test for the custom header:

`testHeader(t, r, "Accept", mediaTypeOrgMembershipPreview)`

Comment (kucuny) [inline github/orgs_teams_test.go]: I added here:
https://github.com/google/go-github/pull/504/files#diff-89cde0e99c2791f7ef5e841a0e6bdf8bR510

**Comment by kucuny** [comment]
@gmlewis Thanks for review! I modified done. :)

**Comment by gmlewis** [comment]
Thank you, @kucuny and @shurcooL!
LGTM.
Merging.

**Comment by gmlewis** [comment]
Included in https://github.com/google/go-github/commit/596563d28d4d7b6cc3f6d5d7fce27131d0715e47

**Pull request closed without merging**